src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/IdnElement.cs

```diff
@@ -50,11 +50,9 @@ public override bool CanConvertFrom(ITypeDescriptorContext context, Type sourceT
 
             public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, object value)
             {
-                string s = value as string;
-                if (s != null)
+                if (value is string s)
                 {
-                    s = s.ToLower(CultureInfo.InvariantCulture);
-                    switch (s)
+                    switch (s.ToLowerInvariant())
```

Comment:
Is this allocation better than three `Equals(..., StringComparison.InvariantCultureIgnoreCase)` calls?

Reply:
This would have been a behavioral change. Had they used `ToUpperInvariant` instead of `ToLowerInvariant`, then using `Equals(..., OrdinalIgnoreCase)` would behave the same.

Reply:
Why is it a behavioral change?

Reply:
It's described in a little more detail at https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1308, but the short of it is that there are some inputs for which `a.ToUpperInvariant() == b.ToUpperInvariant()` but `a.ToLowerInvariant() != b.ToLowerInvariant()`. On Windows, an `OrdinalIgnoreCase` comparison normalizes to uppercase before performing the comparison. Linux today does the same thing, but I opened the issue https://github.com/dotnet/runtime/issues/26961 to reconsider this behavior since the Unicode standard does not recommend doing this.

Reply:
> some inputs

What about for these three specific strings, though? There are inputs that would return true when compared with these three strings when ignoring case invariantly but that would return false when lowercased and then compared? What's an example?

Reply:
For these three specific strings, at this specific moment in time, nothing. However:

 * If https://github.com/dotnet/runtime/pull/27540 goes through as-is, `'İ'` will lowercase to `'i'` when running on Linux (ICU).
 * Certain other non-ASCII characters already lowercase to ASCII characters. For instance, under ICU today, _three_ total characters (`'k'`, `'K'`, and `'K' (U+212A KELVIN SIGN)`) all lowercase to `'k'`. However, only _two_ characters (`'k'` and `'K'`) uppercase to `'K'`. I realize this isn't one of the characters in these strings, but it does demonstrate that "normalize to uppercase" and "normalize to lowercase" are behaviorally inconsistent.
 * There's nothing stopping Unicode from tomorrow introducing a new character that lowercases to one of the ASCII characters contained within these three strings.

The risk is probably low enough where it's not worth worrying about. But with this PR I was trying to avoid introducing behavioral changes beyond "this uses a culture-aware conversion but it really should have been non-linguistic instead."

Reply:
Ok.  It's fine, thanks.